packages/playground/src/weblets/tf_openwebui.vue

```diff
@@ -0,0 +1,281 @@
+const gridStore = useGrid();
+const grid = gridStore.client as GridClient;
+
+watch(
+  [dedicated, rentedByMe],
+  ([dedicated, rentedByMe]) => {
+    if (dedicated === false && rentedByMe === false) {
+      hasGPU.value = dedicated;
+    }
+  },
+  { immediate: true },
+);
+
+watch(
+  hasGPU,
+  hasGPU => {
+    if (hasGPU) {
+      dedicated.value = true;
+      rentedByMe.value = true;
+    }
+  },
+  { immediate: true },
+);
+
+function finalize(deployment: any, domain: string) {
+  layout.value.reloadDeploymentsList();
+  layout.value.setStatus(
+    "success",
+    `Successfully deployed an Open WebUI instance. The domain URL is "${domain}". You might need to wait a couple of minutes for the installation to complete, e.g. if you see Bad Gateway when opening the webpage, simply wait and refresh the page.`,
+  );
+  layout.value.openDialog(deployment, deploymentListEnvironments.openwebui);
+}
+async function deploy() {
+  layout.value.setStatus("deploy");
+
+  const projectName = ProjectName.Openwebui.toLowerCase() + "/" + name.value;
+
+  const subdomain = getSubdomain({
+    deploymentName: name.value,
+    projectName,
+    twinId: profileManager.profile!.twinId,
+  });
+
+  const domain = selectionDetails.value!.domain!.enabledCustomDomain
+    ? selectionDetails.value!.domain!.customDomain
+    : subdomain + "." + selectionDetails.value!.domain!.selectedDomain?.publicConfig.domain;
+
+  let vm: any;
+  try {
+    layout.value?.validateSSH();
+    updateGrid(grid, { projectName });
+
+    await layout.value.validateBalance(grid!);
+
+    vm = await deployVM(grid!, {
+      name: name.value,
+      machines: [
+        {
+          name: name.value,
+          cpu: solution.value.cpu,
+          memory: solution.value.memory,
+          flist: flist.value!.value,
+          entryPoint: flist.value!.entryPoint,
+          disks: [...disks.value],
+          publicIpv4: ipv4.value,
+          publicIpv6: ipv6.value,
+          planetary: planetary.value,
+          mycelium: mycelium.value,
+          envs: [
+            { key: "SSH_KEY", value: selectedSSHKeys.value },
+            { key: "OPENWEBUI_DOMAIN", value: domain },
+          ],
+          rootFilesystemSize: rootFilesystemSize.value,
+          hasGPU: hasGPU.value,
+          nodeId: selectionDetails.value?.node?.nodeId,
+          gpus: hasGPU.value ? selectionDetails.value?.gpuCards.map(card => card.id) : undefined,
+          rentedBy: rentedBy.value,
+          certified: certified.value,
+        },
+      ],
+      network: {
+        addAccess: wireguard.value || selectionDetails.value!.domain!.enableSelectedDomain,
+        accessNodeId: selectionDetails.value?.domain?.selectedDomain?.nodeId,
+      },
+    });
+  } catch (e) {
+    layout.value.setStatus("failed", normalizeError(e, "Failed to deploy an Open WebUI instance."));
+  }
+
+  if (selectionDetails.value!.domain!.enabledCustomDomain && ipv4.value) {
```

Comment:
```suggestion
  if (!selectionDetails.value?.domain?.enableSelectedDomain) {
```

this is shorter and consist with other solutions